atest/testdata/running/for.robot

```diff
@@ -317,6 +324,120 @@ For In Range With Expressions Containing Floats
     \    @{result} =    Create List    @{result}    ${i}
     Should Be True    @{result} == [3.14, 1.14, -0.86]
 
+For In Zip
+    @{items}=    Create List    a    b    c    d
+    @{things}=    Create List    e    f    g    h
+    : FOR    ${item}    ${thing}    IN ZIP    ${items}    ${things}
+    \    @{result} =    Create List    @{result}    ${item}:${thing}
+    Should Be True    @{result} == ['a:e', 'b:f', 'c:g', 'd:h']
+
+For In Zip With Uneven Lists
+    [Documentation]    Handling lists with different number of elements.  This will ignore any elements after the shortest list ends, just like with Python's zip()
```

Comment:
Longish docs like this could be split to multiple lines by using `...`. The first sentence could also be removed because it tells pretty much what the test name tells.